core-plugins/src/test/java/io/cdap/plugin/batch/source/FileBatchSourceTest.java

```diff
@@ -374,6 +373,62 @@ public void testPathGlobbing() throws Exception {
     Assert.assertTrue(outputValue.contains("CDAP,Platform"));
   }
 
+  @Test
+  public void testSkipHeader() throws Exception {
+    Schema schema = Schema.recordOf("user",
+                                    Schema.Field.of("body", Schema.of(Schema.Type.STRING)),
+                                    Schema.Field.of("offset", Schema.of(Schema.Type.LONG)),
+                                    Schema.Field.of("file", Schema.nullableOf(Schema.of(Schema.Type.STRING))));
+    testSkipHeader("text", "", schema);
+
+    schema = Schema.recordOf("user",
+                             Schema.Field.of("val1", Schema.of(Schema.Type.INT)),
+                             Schema.Field.of("val2", Schema.of(Schema.Type.INT)),
+                             Schema.Field.of("val3", Schema.of(Schema.Type.INT)),
+                             Schema.Field.of("file", Schema.nullableOf(Schema.of(Schema.Type.STRING))));
+    testSkipHeader("csv", ",", schema);
+    testSkipHeader("tsv", "\t", schema);
+    testSkipHeader("delimited", " ", schema);
+
+  }
+
+  private void testSkipHeader(String format, String delimeter, Schema schema) throws Exception {
+    File inputFile = temporaryFolder.newFile();
+
+    try (Writer writer = new FileWriter(inputFile)) {
+      writer.write(Joiner.on(delimeter).join(new String[] {"a", "b", "c"}) + "\n");
+      writer.write(Joiner.on(delimeter).join(new String[] {"1", "2", "3"}) + "\n");
+      writer.write(Joiner.on(delimeter).join(new String[] {"4", "5", "6"}) + "\n");
+    }
+
+    String outputDatasetName = UUID.randomUUID().toString();
+    String appName = UUID.randomUUID().toString();
+
+    String uri = inputFile.toURI().toString();
+    ApplicationManager appManager = createSourceAndDeployApp(appName, inputFile, format, outputDatasetName, schema,
+                                                             false, delimeter, true);
+
+    appManager.getWorkflowManager(SmartWorkflow.NAME).startAndWaitForRun(ProgramRunStatus.COMPLETED,
+                                                                         5, TimeUnit.MINUTES);
+
+    Set<StructuredRecord> expected = null;
+
+    if (format.equals("text")) {
```

Comment:
changing behavior based on some input argument is usually something to avoid. I think it's better to have this method return the output and have the caller assert equality.

Reply:
Fixed